src/Ztm.Threading.Tests/TimerSchedulers/ThreadPoolSchedulerTests.cs

```diff
@@ -97,26 +97,34 @@ public void Schedule_WithNullPeriod_ShouldOneShot()
             }
         }
 
-        [Fact(Skip = "This test is not reliable due to it need to assume that timer has been elapsed")]
-        public async Task Schedule_WithNonNullPeriod_ShouldNotOneShot()
+        [Fact]
+        public void Schedule_WithNonNullPeriod_ShouldNotOneShot()
         {
-            using (var elapsed = new ManualResetEventSlim())
+            using (var cde = new CountdownEvent(2))
             {
                 // Arrange.
                 var context = new object();
 
                 this.handler.Setup(f => f(It.IsAny<object>()))
-                            .Callback<object>(c => elapsed.Set());
+                            .Callback<object>(c =>
+                            {
+                                lock(cde)
```

Comment:
We should not do this due to `CountdownEvent` is already thread-safe. We should invoke `Signal` directly and ignore `InvalidOperationException` instead.